projects/ngx-avatar/src/lib/sources/vkontakte.ts

```diff
@@ -24,12 +24,15 @@ export class Vkontakte extends AsyncSource {
   /**
    * extract vkontakte avatar from json data
    */
-  public processResponse(data: any): string {
+  public processResponse(data: any): string | null {
     // avatar key property is the size used to generate avatar url
     // size property is always the last key in the response object
     const sizeProperty = Object.keys(data['response'][0]).pop();
-    // return avatar src
-    return data['response'][0][sizeProperty];
+    if (!sizeProperty) {
+      return null;
+    }
+
+    return data['response'][0][sizeProperty] || null;
```

Comment:
```suggestion
    return data['response'][0][sizeProperty];
```

Maybe we don't need this extra null fallback as we already have a new check before the return.

Reply:
Correct, I pushed a change.